mindsdb/integrations/handlers/notion_handler/notion_handler.py

```diff
@@ -0,0 +1,234 @@
+import os
+import time
+import pandas as pd
+from collections import defaultdict
+
+from pydantic import BaseModel, Extra
+from notion_client import Client
+
+from mindsdb.utilities import log
+from mindsdb.utilities.config import Config
+from mindsdb.integrations.libs.api_handler import APIHandler
+from mindsdb.integrations.libs.response import HandlerStatusResponse as StatusResponse
+
+from .notion_table import (
+    NotionBlocksTable,
+    NotionCommentsTable,
+    NotionDatabaseTable,
+    NotionPagesTable,
+)
+
+
+class NotionHandlerArgs(BaseModel):
+    target: str = None
+    ft_api_info: dict = None
+    ft_result_stats: dict = None
+    runtime: str = None
+    api_token: str = None
+
+    class Config:
+        # for all args that are not expected, raise an error
+        extra = Extra.forbid
+
+
+class NotionHandler(APIHandler):
+    name = "notion"
+
+    def __init__(self, name: str, **kwargs):
+        """constructor
+        Args:
+            name (str): the handler name
+        """
+        super().__init__(name)
+
+        args = kwargs.get("connection_data", {})
```

Comment:
You don't need this logic either. Let's simply do,
`self.connection_args = kwargs.get("connection_data", {})`

Let's remove the associated logic (that is no longer necessary) as well.